src/connection/host_lookup.c

```diff
@@ -0,0 +1,435 @@
+	{
+	  return NULL;
+	}
+
+    }
+
+/*Look up in the hostent_List*/
+  for (i = 0; i < host_conf_element_Count; i++)
+    {
+      if ((lookup_case == HOSTNAME_TO_IPADDR) && (strcmp (hostname, hostent_List[i].hostname) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+
+      if ((lookup_case == IPADDR_TO_HOSTNAME) && (strcmp (addr_transform_buf, hostent_List[i].ipaddr) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+    }
+
+  if (find_index < 0)
+    {
+      return NULL;
+//this should be set to error message
+    }
+
+  if (inet_pton (AF_INET, hostent_List[find_index].ipaddr, addr_transform_buf2) < 1)
+    {
+      return NULL;
+    }
+
+  /*set the hostent struct for the return value */
+  hp.h_addrtype = AF_INET;
+  hp.h_length = 4;
+  hp.h_name = (char *) malloc (sizeof (char) * HOSTNAME_BUF_SIZE);
+  hp.h_aliases = (char **) malloc (sizeof (char *));
+  hp.h_aliases[0] = (char *) malloc (sizeof (char) * HOSTNAME_BUF_SIZE);
+  hp.h_addr_list = (char **) malloc (sizeof (char *) * HOSTNAME_BUF_SIZE);
+  hp.h_addr_list[0] = (char *) malloc (sizeof (char) * IPADDR_LEN);
+
+  strcpy (hp.h_name, hostent_List[find_index].hostname);
+  strcpy (hp.h_aliases[0], hostent_List[find_index].hostname);
+  memcpy (hp.h_addr_list[0], addr_transform_buf2, sizeof (hp.h_addr_list[0]));
+
+  return &hp;
+}
+
+static int
+host_conf_load ()		//set hash table to discover error
+{
+  FILE *fp;
+  char file_line[HOSTNAME_BUF_SIZE + 1];
+  char line_buf[HOSTNAME_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token;
+  char *save_ptr_strtok;
+  /*delimiter */
+  char *delim = " \t\n";
+
+  int host_count = 0;
+
+  /*False, if hostent_List[index].hostname is set */
+  bool storage_flag = IPADDR_STORE;
+
+  memset (file_line, 0, HOSTNAME_BUF_SIZE + 1);
+  memset (line_buf, 0, HOSTNAME_BUF_SIZE + 1);
+
+  host_count = host_conf_element_Count;
+  assert (!host_count);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, "hosts.conf");
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+      return -1;
+      //error msg-> temp
+    }
+
+  while (fgets (file_line, HOSTNAME_BUF_SIZE, fp) != NULL)
+    {
+//verifying duplicated hostname will be implemented soon
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      storage_flag = IPADDR_STORE;
+
+      char map_ipaddr[HOSTNAME_BUF_SIZE];
+      char map_hostname[HOSTNAME_BUF_SIZE];
+
+      do
+	{
+	  if (*token == '\0' || *token == '#')
+	    {
+	      break;
+	    }
+	  if (!storage_flag)
+	    {
+	      strcpy (hostent_List[host_count].ipaddr, token);
+	      strcpy (map_ipaddr, token);
+
+	      storage_flag = HOSTNAME_STORE;
+	    }
+	  else
+	    {
+	      strcpy (hostent_List[host_count].hostname, token);
+	      strcpy (map_hostname, token);
```

Comment:
If the length of token is greater than hostent_List[host_count].hostnameand map_hostname, overflow may occur.
And, shouldn't map_hostname and hostent_List[host_count].hostname have the same length?

Reply:
1. I will fix that over 256 hostname and 16 ip address is considered as wrong name.
2. the length of map_ipaddr will be fixed IPADDR_LEN(16+1)